### PR TITLE
muc: add a add to contacts button

### PR DESCRIPTION
**Pull request by sloonz**
Pull request for #3061

### CODE REVIEW

File: src/plugins/muc-views/modals/occupant.js [hunk 1]
```diff
@@ -36,13 +36,23 @@ export default class OccupantModal extends BaseModal {
         const role = this.model?.get('role');
         const affiliation = this.model?.get('affiliation');
         const hats = this.model?.get('hats')?.length ? this.model.get('hats') : null;
-        return tpl_occupant_modal({ jid, vcard, nick, occupant_id, role, affiliation, hats });
+        const muc = this.model.collection.chatroom;
+        const addToContacts = muc.features.get('nonanonymous') && jid && jid != _converse.bare_jid ? this.addToContacts.bind(this) : null;
```
Comment (jcbrand) [inline src/plugins/muc-views/modals/occupant.js:40]: This doesn't check whether the user isn't already in your roster.

You can check like this:
```javascript
const contact = await api.contacts.get(jid);
```

Comment (sloonz) [reply src/plugins/muc-views/modals/occupant.js:40]: is renderModal allowed to be async ?

Comment (jcbrand) [reply src/plugins/muc-views/modals/occupant.js:40]: Good question. The answer is "no" :)

What you can do however is pass a promise and use the `until` directive to render nothing until the promise has resolved.

https://lit.dev/docs/templates/directives/#until

**Comment by jcbrand** [comment]
Thanks :)

**Pull request merged**